### PR TITLE
Commerce 9507 Cannot delete product with versioning

### PR DESCRIPTION
**Pull request by christian-chiappa**

### CODE REVIEW

**Comment by christian-chiappa** [comment]
ci:test:sf

**Comment by christian-chiappa** [comment]
ci:test:sf

**Comment by christian-chiappa** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java [hunk 1]
```diff
@@ -665,7 +670,7 @@ public void deleteCPInstances(long cpDefinitionId) throws PortalException {
 			cpInstancePersistence.findByCPDefinitionId(cpDefinitionId);
 
 		for (CPInstance cpInstance : cpInstances) {
-			cpInstanceLocalService.deleteCPInstance(cpInstance);
+			cpInstanceLocalService.deleteCPInstance(cpInstance, false);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java:673]: I would expect to have the parameter makeCopy in this method and call pass false in the deleteCPDefinition also for SKUs

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java:673]: Also becasue this method could be called elsewhere

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
the commit named test must be reorganized because it contains a build service for sure and a change to a jsp that should not be part of this ticket.

**Pull request closed without merging**